File: internal/dto/device.go
package dto

import "time"

type CreateDeviceRequest struct {
	Hostname string `json:"hostname" binding:"required,min=1,max=255"`
	IP       string `json:"ip" binding:"required,ip"`
	Location string `json:"location" binding:"required"`
}

type UpdateDeviceRequest struct {
	Hostname *string `json:"hostname" binding:"omitempty,min=1,max=255"`
	IP       *string `json:"ip" binding:"omitempty,ip"`
	Location *string `json:"location"`
	IsActive *bool   `json:"is_active"`
}

type DeviceResponse struct {
	ID        uint      `json:"id"`
	Hostname  string    `json:"hostname"`
	IP        string    `json:"ip"`
	Location  string    `json:"location"`
	IsActive  bool      `json:"is_active"`
	CreatedAt time.Time `json:"created_at"`
}

File: frontend/src/types/device.ts
export interface Device {
  id: number;
  hostname: string;
  ip: string;
  location: string | null;
  is_active: boolean;
  created_at: string;
}

export interface CreateDeviceRequest {
  hostname: string;
  ip: string;
  location?: string;
}

export interface UpdateDeviceRequest {
  hostname?: string;
  ip?: string;
  location?: string;
  is_active?: boolean;
}
